presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -98,50 +97,40 @@ public CachingHiveMetastore(
-    }
-
-    public CachingHiveMetastore(
-            ExtendedHiveMetastore delegate,
-            ExecutorService executor,
-            Duration cacheTtl,
-            Duration refreshInterval,
-            long maximumSize,
-            boolean isPartitioningEnabled)
-    {
-        this(
-                delegate,
-                executor,
-                OptionalLong.of(cacheTtl.toMillis()),
-                refreshInterval.toMillis() >= cacheTtl.toMillis() ? OptionalLong.empty() : OptionalLong.of(refreshInterval.toMillis()),
-                maximumSize,
-                isPartitioningEnabled);
+                CachingHiveMetastoreConfig.builder()
+                        .setAllCacheTtl(metastoreClientConfig.getMetastoreCacheTtl())
+                        .setAllCacheRefreshInterval(metastoreClientConfig.getMetastoreRefreshInterval())
+                        .setAllCacheMaxSize(metastoreClientConfig.getMetastoreCacheMaximumSize())
+                        .setPartitionVersioningEnabled(metastoreClientConfig.isPartitionVersioningEnabled())
+                        .setPartitionCacheTtl(metastoreClientConfig.getMetastorePartitionCacheTtl())
+                        .setPartitionCacheRefreshInterval(metastoreClientConfig.getMetastorePartitionRefreshInterval())
+                        .setPartitionCacheMaxSize(metastoreClientConfig.getMetastorePartitionCacheMaximumSize())
+                        .build());
     }
 
     public static CachingHiveMetastore memoizeMetastore(ExtendedHiveMetastore delegate, long maximumSize)
     {
         return new CachingHiveMetastore(
                 delegate,
                 newDirectExecutorService(),
-                OptionalLong.empty(),
-                OptionalLong.empty(),
-                maximumSize,
-                false);
+                CachingHiveMetastoreConfig.builder().setAllCacheMaxSize(maximumSize).setPartitionVersioningEnabled(false).build());
     }
 
-    private CachingHiveMetastore(
+    public CachingHiveMetastore(
             ExtendedHiveMetastore delegate,
             ExecutorService executor,
-            OptionalLong expiresAfterWriteMillis,
-            OptionalLong refreshMills,
-            long maximumSize,
-            boolean isPartitioningEnabled)
+            CachingHiveMetastoreConfig cacheConfig)
     {
         this.delegate = requireNonNull(delegate, "delegate is null");
         requireNonNull(executor, "executor is null");
+        requireNonNull(cacheConfig, "cacheConfig is null");
+        this.isPartitionVersioningEnabled = cacheConfig.isPartitionVersioningEnabled();
+        OptionalLong expiresAfterWriteMillis = cacheConfig.getAllCacheTtl();
+        OptionalLong refreshMills = cacheConfig.getAllCacheRefreshInterval();
+        long maximumSize = cacheConfig.getAllCacheMaxSize();
+        OptionalLong partitionExpireAfterWriteMillis = isPartitionVersioningEnabled ? cacheConfig.getPartitionCacheTtl() : expiresAfterWriteMillis;
```

Comment:
I think we should just call `cacheConfig. getPartitionCacheTtl()` regardless of `isPartitionVersioningEnabled`?